### ISSUE TITLE
expand `colorPalette` flexibility

### ISSUE THREAD

**Issue by VinSpee** [chakra-ui/panda#1062]
### Description

expand the flexibility of the Virtual Color feature by allowing for un-opinionated nested object definitions

### Problem Statement/Justification

Hi! I've been using panda in two large projects. As a former Chakra user and a tailwind tolerator, I was delighted to hop on the panda train as quickly as possible and guide my teams in that direction.

One idea I really like is [Virtual Color](https://panda-css.com/docs/guides/virtual-color).

I like to build component libraries in a very flexible way, and a technique I use is to provide `aesthetic` and `color` props. In panda, I model it like this:

```ts
const config = defineConfig({
	theme: {
		tokens: {
			blue: {
				50: {
					value: 'blue'
				},
				900: {
					value: 'cyan'
				}
			},
			red: {
				50: {
					value: 'red'
				},
				900: {
					value: 'magenta'
				}
			}
		}
	}
});

const buttonRecipe = defineRecipe({
	name: 'button',
	base: {
		…baseProps
	},
	variants: {
		color: {
			blue: {
				colorPalette: 'blue',
			},
			red: {
				colorPalette: 'red',
			}
		},
		aesthetic: {
			solid: {
				backgroundColor: 'colorPalette.50'
				color: 'colorPalette.900'
			}
		}
	}
});
```

This is straightforward enough.

However, in my current project, we're taking an approach where we lean into semantic tokens, with a theme configuration like so:

```ts
const config = defineConfig({
	theme: {
		semanticTokens: {
			core: {
				light: {
					DEFAULT: {
						value: 'blue'
					},
					accent: {
						value: 'cyan'
					}
			},
			brand: {
				light: {
					DEFAULT: {
						value: 'red',
					},
					accent: {
						value: 'magenta'
					}
				}
			}
		}
	}
})
```

which means I want to define my button like this:

```ts
const buttonRecipe = defineRecipe({
	name: 'button',
	base: {
		…baseProps
	},
	variants: {
		color: {
			core: {
				colorPalette: 'core',
			},
			brand: {
				colorPalette: 'brand',
			}
		},
		aesthetic: {
			solid: {
				backgroundColor: 'colorPalette.light'
				color: 'colorPalette.light.accent'
			}
		}
	}
});
```

This doesn't work.

### Proposed Solution or API

I see [in the docs that `colorPalette` requires 50–900 keys](https://panda-css.com/docs/guides/virtual-color#:~:text=%F0%9F%92%A1-,Important%3A%20A%20valid%20colorPalette%20value%20is%20an%20color%20object%20that%20has%2050%20to%20900%20keys.,-const%20className%20%3D).

1. Why does `colorPalette` have an opinion about its shape?
2. Would it be reasonable to expand the flexibility of Virtual Colors to allow for nested tokens?

### Alternatives

- altering my color palette to use `50`-`900`
- not using `colorPalette` or Virtual Color at all

### Additional Information

Here's a failing case: https://play.panda-css.com/XIIQGvJhd9

Thanks for the excellent work on the project – it makes building UIs so fun, and is the most lovely successor to chakra, which I loved for a long time.

**Comment by isBatak** [comment]
It seems like it mostly works, only the hover state is not detected 🤔
https://play.panda-css.com/ebvVxk8H8U

**Comment by VinSpee** [comment]
> @VinSpee 
> 
> It seems like it mostly works, only the hover state is not detected 🤔
> 
> https://play.panda-css.com/ebvVxk8H8U

Exactly. The nested case fails (has nothing to do with the hover state, it's due to that token being nested an additional level)
